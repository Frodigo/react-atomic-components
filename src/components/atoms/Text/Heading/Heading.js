import React from 'react';
import PropTypes from 'prop-types';

/**
 * Heading wrapper.
 * Please add `inverse` class if you want to show heading with white color.
 */
const Heading = (props) => {
    const { children, type, classes, textAlign } = props;

    let heading;

    const styles = {
        textAlign
    };

    switch (type) {
        case 'h1': {
            heading = (
                <h1 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h1>
            );

            break;
        }

        case 'h2': {
            heading = (
                <h2 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h2>
            );

            break;
        }

        case 'h3': {
            heading = (
                <h3 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h3>
            );

            break;
        }

        case 'h4': {
            heading = (
                <h4 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h4>
            );

            break;
        }

        case 'h5': {
            heading = (
                <h5 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h5>
            );

            break;
        }

        case 'h6': {
            heading = (
                <h6 className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </h6>
            );

            break;
        }

        default: {
            heading = (
                <div className={`${classes ? classes : ''}`} style={styles}>
                    {children}
                </div>
            );

            break;
        }
    }

    return <>{heading}</>;
};

Heading.propTypes = {
    children: PropTypes.string,
    classes: PropTypes.string,
    textAlign: PropTypes.string,
    type: PropTypes.string.isRequired
};

export default Heading;
