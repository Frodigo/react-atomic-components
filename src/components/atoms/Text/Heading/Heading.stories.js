import React from 'react';

import Heading from './Heading';

export default {
    title: 'Atoms/Headings',
    component: Heading,
    argTypes: {
        type: {
            control: {
                type: 'select',
                options: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
            }
        },
        textAlign: {
            control: {
                type: 'select',
                options: ['left', 'center', 'right']
            }
        }
    }
};

const Template = (args) => <Heading {...args} />;

export const HeadingDefault = Template.bind({});
HeadingDefault.args = {
    type: 'h1',
    children: 'Heading'
};

const InverseTemplate = (args) => (
    <div
        style={{
            backgroundColor: 'black',
            padding: `${30}px`
        }}
    >
        <Heading {...args} />
    </div>
);

export const HeadingInverse = InverseTemplate.bind({});
HeadingInverse.args = {
    type: 'h1',
    children: 'Heading',
    classes: 'inverse'
};
