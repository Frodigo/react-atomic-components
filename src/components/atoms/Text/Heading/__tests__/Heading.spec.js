import React from 'react';
import { render } from '@testing-library/react';

import Heading from '../Heading';

describe('Heading should be rendered', () => {
    test('Heading 1 renders correctly', () => {
        const { container } = render(<Heading type={'h1'}>Heading 1</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading 2 renders correctly', () => {
        const { container } = render(<Heading type={'h2'}>Heading 2</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading 3 renders correctly', () => {
        const { container } = render(<Heading type={'h3'}>Heading 3</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading 4 renders correctly', () => {
        const { container } = render(<Heading type={'h4'}>Heading 4</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading 5 renders correctly', () => {
        const { container } = render(<Heading type={'h5'}>Heading 5</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading 6 renders correctly', () => {
        const { container } = render(<Heading type={'h6'}>Heading 6</Heading>);

        expect(container).toMatchSnapshot();
    });

    test('Heading inverse should render correctly', () => {
        const { container } = render(
            <Heading type="h1" classes="inverse">
                Heading inverse
            </Heading>
        );

        expect(container).toMatchSnapshot();
        expect(container.firstChild.classList.contains('inverse')).toBe(true);
    });

    test('Heading with specific text align should render correctly', () => {
        const { container } = render(
            <Heading type="h3" textAlign="center">
                Heading centered
            </Heading>
        );

        expect(container).toMatchSnapshot();
        expect(container.firstChild.style.textAlign).toBe('center');
    });
});
