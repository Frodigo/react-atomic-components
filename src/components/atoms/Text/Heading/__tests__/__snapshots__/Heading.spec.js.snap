// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should be rendered Heading 1 renders correctly 1`] = `
<div>
  <h1
    class=""
  >
    Heading 1
  </h1>
</div>
`;

exports[`Heading should be rendered Heading 2 renders correctly 1`] = `
<div>
  <h2
    class=""
  >
    Heading 2
  </h2>
</div>
`;

exports[`Heading should be rendered Heading 3 renders correctly 1`] = `
<div>
  <h3
    class=""
  >
    Heading 3
  </h3>
</div>
`;

exports[`Heading should be rendered Heading 4 renders correctly 1`] = `
<div>
  <h4
    class=""
  >
    Heading 4
  </h4>
</div>
`;

exports[`Heading should be rendered Heading 5 renders correctly 1`] = `
<div>
  <h5
    class=""
  >
    Heading 5
  </h5>
</div>
`;

exports[`Heading should be rendered Heading 6 renders correctly 1`] = `
<div>
  <h6
    class=""
  >
    Heading 6
  </h6>
</div>
`;

exports[`Heading should be rendered Heading inverse should render correctly 1`] = `
<div>
  <h1
    class="inverse"
  >
    Heading inverse
  </h1>
</div>
`;

exports[`Heading should be rendered Heading with specific text align should render correctly 1`] = `
<div>
  <h3
    class=""
    style="text-align: center;"
  >
    Heading centered
  </h3>
</div>
`;
